### PR TITLE
Make science ghost a mob critter

### PR DESCRIPTION
**Pull request by FlameArrow57**
[AI][INTERNAL][REWORK]

## About the PR 
This PR just converts /obj/critter/aberration to /mob/living/critter/aberration.

Also adds some functionality to make them possessable and act the same as when AI controlled.

## Why's this needed? 
Object critter removal

### CODE REVIEW

File: code/mob/living/critter/aberration.dm [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+	projCanHit(datum/projectile/P)
+		return P.damage_type == D_ENERGY
+
+	do_disorient(stamina_damage, weakened, stunned, paralysis, disorient = 60, remove_stamina_below_zero = 0, target_type = DISORIENT_BODY, stack_stuns = 1)
+		return
+
+	TakeDamage(zone, brute, burn, tox, damage_type, disallow_limb_loss)
+		if (brute >= 0)
+			return
```
Comment (Tarmunora) [inline code/mob/living/critter/aberration.dm:104]: pointless conditional

**Review by Tarmunora — CHANGES_REQUESTED** [review]
One general concern - science ghosts should have the ability to trap people if the victim corners themself - however, I'm concerned that mobification could defang that if not checked for (rest-sliding out, disarm-shove, etc.).

File: code/mob/living/critter/aberration.dm (hunk 1 above)
Comment (FlameArrow57) [reply code/mob/living/critter/aberration.dm:104]: Sure, it was since healing does TakeDamage(-amt) if wanted for anything

**Comment by FlameArrow57** [comment]
attack_hand() returning without calling parent prevents them from being disarm shoved, was able to fix them being slid under. Not aware of any other cases

**Comment by FlameArrow57** [comment]
Was that for that running into one won't push it back too?

File: code/mob/living/critter/aberration.dm [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+	blob_act(power)
+		return
+
+	proc/hurt(damage)
+		var/datum/healthHolder/Br = src.get_health_holder("brute")
+		Br?.TakeDamage(damage)
+
+	seek_target(range)
+		var/targets = ..(7)
```
Comment (amylizzle) [inline code/mob/living/critter/aberration.dm:116]: looping through twice seems inefficient. It'd be better to just copy the parent proc and add an extra continue for the crown.

File: code/mob/living/critter/aberration.dm [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+		for (var/mob/living/carbon/human/H in targets)
+			if (istype(H.head, /obj/item/clothing/head/void_crown))
+				targets -= H
+
+		. = targets
+
+	Cross(atom/movable/mover)
+		if (!istype(mover, /mob))
+			return ..()
+		var/mob/M = mover
```
Comment (amylizzle) [inline code/mob/living/critter/aberration.dm:126]: Are there any conditions where a mob *can* move through a science ghost? Might be better to just replace all this with
 ```
else
    return FALSE
```

**Review by amylizzle — CHANGES_REQUESTED** [review]
AI looks fine, some optimisations on the mob code though.

File: code/mob/living/critter/aberration.dm (hunk 3 above)
Comment (FlameArrow57) [reply code/mob/living/critter/aberration.dm:126]: Small mobs like mice? Dunno

**Comment by Tarmunora** [comment]
> Was that for that running into one won't push it back too?

setting anchored will prevent being pushed around/pulled/etc

**Comment by FlameArrow57** [comment]
Okay, seems to work fine

**Review by Tarmunora — APPROVED** [review]

**Pull request merged**